### PR TITLE
Use template file for adding table functions grammar

### PR DESCRIPTION
**Pull request by rohangarg**

### CODE REVIEW

**Comment by rohangarg** [comment]
@paul-rogers @imply-cheddar : This proposes an alternate way to extend the `FROM` grammar in Calcite. The new changes merged recently for extended table functions use a script `edit-parser` to update the parser file and added the new production rules in it. While technically currently that is the only way to extend `FROM` grammar in calcite for now, the script method is a bit difficult to use for the following reasons : 
1. Doing more complex operations/edits on the grammar is a bit tricky in the script
2. Getting a whole view of the extended grammar from code is a bit hard since the script generates the new parser dynamically
3. Since the update is dynamic, every update requires a search and replace which makes the update a bit fragile and susceptible to logical problems when the underlying grammar updates

An alternative approach tried in this change is to pull the relevant production rules from the calcite grammar and have them as template files which can be edited as per druid's usecases. This provides full control over the pulled production rule, a simple full view of the augmented grammar and also lesser search-replace operations (one in the current case in the pom file). 

Please let me know your thoughts on it!

**Comment by rohangarg** [comment]
closing this since better to have discussion on #13553

File: sql/src/main/codegen/includes/from.ftl [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+                 tableRef = ParameterizeOperator.PARAM.createCall(tableRef, paramList);
+             }
+         ]
+         [
+             [ <EXTEND> ]
+             extendList = ExtendList() {
+                 tableRef = extend(tableRef, extendList);
+             }
+         ]
+         over = TableOverOpt() {
```
Comment (paul-rogers) [inline sql/src/main/codegen/includes/from.ftl:206]: Remove the following lines:
```text
         [
             [ <EXTEND> ]
             extendList = ExtendList() {
                 tableRef = extend(tableRef, extendList);
             }
         ]
```

Druid supports `EXTEND` now, but not in the same place as Phoenix. Removing these lines avoids the need for us to parse, then later reject, the Phoenix syntax.

**Review by paul-rogers — APPROVED** [review]
@rohangarg, thanks for this cleanup. I verified that that the new `from.ftl` contains the same code that was previously produced by `edit-parser.py`. Found one difference, see the note.

Otherwise, LGTM. Approving, but please do apply that missing change.

**Comment by paul-rogers** [comment]
The one draw-back to the copy-and-edit approach is that each Calcite upgrade requires a manual step to match up our version against the new Calcite version, and to understand the difference between any Calcite changes and our own. The edit script was meant to automate that process. But, it is fine to make it a manual process, since we seldom upgrade Calcite. (Note, however, that there is an open PR to do just that: to upgrade Calcite.)

File: sql/src/main/codegen/includes/from.ftl (hunk 1 above)
Comment (rohangarg) [reply sql/src/main/codegen/includes/from.ftl:200]: Thanks for catching this! Fixed

**Comment by rohangarg** [comment]
> The one draw-back to the copy-and-edit approach is that each Calcite upgrade requires a manual step to match up our version against the new Calcite version, and to understand the difference between any Calcite changes and our own. The edit script was meant to automate that process. But, it is fine to make it a manual process, since we seldom upgrade Calcite. (Note, however, that there is an open PR to do just that: to upgrade Calcite.)

I was thinking that manual checking of conflicts also provides value since it would help in catching them ahead of time instead of logical conflicts staying the code and failing later.

**Pull request merged**